DEVELOPMENT_GUIDE.md

```diff
@@ -154,6 +154,39 @@ conventions are best practices that we have learnt over time.
 -   Do not catch the broader `Exception`, unless you have a really
     strong reason to do. You must explain the reason in great detail in
     comments.
+    
+    
+Testing
+-------
+
+We need thorough test coverage to ensure the code change works today, 
+and continues to work in future. When you make a code change, use the 
+following framework to decide the kinds of tests to write:
+
+- When you adds/removed/modifies code paths (aka branches/arcs), 
+  write **unit tests** with goal of making sure the flow works. Focus
+  on verifying the flow and use mocks to isolate from as many 
+  external dependencies as you can. "External dependencies" 
+  includes system calls, libraries, other classes/methods you wrote
+  but logically outside of the system-under-test.
+  
+  > Aim to test with complete isolation
+  
+- When your code uses external dependencies, write **functional tests** 
+  to verify some flows by including as many external dependencies as 
```

Comment:
I dont think it should state as many external dependencies, but just explicitly the dependencies that need to be tested.

Reply:
We don't know what these dependencies are here. I provided some examples in later sentence instead